.github/workflows/js_tests.yml

```
@@ -12,27 +12,38 @@ permissions:
   contents: read
 
 jobs:
+  setup_matrix:
+    name: Setup matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.build_matrix.outputs.matrix }}
+    steps:
+      - name: Build test matrix
+        id: build_matrix
+        uses: ekohl/gha-matrix-builder@v0
+        with:
+          base_matrix_url: "https://raw.githubusercontent.com/${{ github.repository }}/${{ github.base_ref }}/.github/matrix.json"
+
   test:
     permissions:
       checks: write  # for coverallsapp/github-action to create new checks
       contents: read  # for actions/checkout to fetch code
     runs-on: ubuntu-latest
+    needs: setup_matrix
     timeout-minutes: 20
     strategy:
       fail-fast: false
-      matrix:
-        node-version:
-          - 14
+      matrix: ${{ fromJson(needs.setup_matrix.outputs.matrix) }}
```

I wonder if we can somehow be smart and only test on the first Ruby version. Running the JS tests with all Ruby versions probably doesn't make sense.
```suggestion
      matrix:
        ruby:
          - ${{ fromJson(needs.setup_matrix.outputs.matrix).ruby[0] }}
        node: ${{ fromJson(needs.setup_matrix.outputs.matrix).node }}
```

But perhaps with webpack 5 the webpack config no longer executes Ruby to determine plugins (not sure, couldn't see it immediately) and it no longer needs Ruby.